nautobot/core/models/querysets.py

```diff
@@ -16,7 +16,25 @@ def count_related(model, field):
     return Coalesce(subquery, 0)
 
 
-class RestrictedQuerySet(QuerySet):
+class CompositeKeyQuerySetMixin:
+    """
+    Mixin to extend a base queryset class with support for filtering by `composite_key=...`.
+    """
+
+    def filter(self, *args, **kwargs):
+        """
+        This is an enhanced version of natural-key slug support from django-natural-keys.
+        Counterpart to BaseModel.composite_key property.
```

Comment:
Is it possible to provide an example in the comment? I don't know what an example of what `composite_key` is going to be, so hard for me to imagine what is going on here.

Reply:
Added documentation - this is mostly a mixin for internal use rather than anything a plugin would use, FWIW (as it's included in the `RestrictedQuerySet` class) but documentation is never a bad thing. :)